src/tpm.rs

```diff
@@ -396,13 +468,51 @@ fn read_file_output_path(output_path: String) -> std::io::Result<String> {
     Ok(contents)
 }
 
+/*
+ * These test are for Centos and tpm4720 elmulator install environment. It
+ * will fail if tpm is not install in the system.
+ */
 #[cfg(test)]
 mod tests {
     use super::*;
+    use std::fs;
+
+    #[test]
+    fn test_read_file_output_path() {
+        assert_eq!(
+            read_file_output_path("test_input.txt".to_string()).unwrap(),
+            "Hello World!\n"
+        );
+    }
 
+    //Comment out for tpm unit testing
```

Comment:
Actual TPM tests are good and we should have them.  Rather than commenting them out, please make running them conditional on whether we have the capability to do so.

Reply:
For the test condition, I found out there are two ways to do. One is to let the problem test the system capability for those command using the same "cargo test" command. The other one is let the user change the running condition. For example, if the system has the tpm run the test 'cargo test --feature 'tpm_ok' ', otherwise run 'cargo test', the tpm test case will not execute with normal cargo test.

This is the implementation of auto command detection. It is ready for review.